File: Solana_Token.ts
import { Account, PublicKey, Transaction, TransactionInstruction } from "@solana/web3.js";
import { TOKEN_PROGRAM_ID, Token, ASSOCIATED_TOKEN_PROGRAM_ID, TOKEN_ACCOUNT_LAYOUT } from "@solana/spl-token";

// Set the network and token account addresses
const network = "https://api.mainnet-beta.solana.com";
const tokenAccountAddress = new PublicKey("<TOKEN_ACCOUNT_ADDRESS>");

// Set the payer account and its associated token account
const payerAccount = new Account("<PAYER_PRIVATE_KEY>");
const associatedTokenAddress = await Token.getAssociatedTokenAddress(
  ASSOCIATED_TOKEN_PROGRAM_ID,
  TOKEN_PROGRAM_ID,
  new PublicKey("<TOKEN_MINT_ADDRESS>"),
  payerAccount.publicKey,
);

// Create the token object
const token = new Token(
  provider.connection,
  new PublicKey("<TOKEN_MINT_ADDRESS>"),
  TOKEN_ACCOUNT_LAYOUT,
  payerAccount,
);

// Create the transaction instructions
const instructions: TransactionInstruction[] = [];

// Create the token account if it doesn't exist
const tokenAccountExists = await provider.connection.getAccountInfo(tokenAccountAddress);
if (!tokenAccountExists) {
  const createAccountInstruction = SystemProgram.createAccount({
    fromPubkey: payerAccount.publicKey,
    newAccountPubkey: tokenAccountAddress,
    lamports: await provider.connection.getMinimumBalanceForRentExemption(TOKEN_ACCOUNT_LAYOUT.span),
    space: TOKEN_ACCOUNT_LAYOUT.span,
    programId: TOKEN_PROGRAM_ID,
  });
  instructions.push(createAccountInstruction);
}

// Create the associated token account if it doesn't exist
const associatedTokenAccountExists = await provider.connection.getAccountInfo(associatedTokenAddress);
if (!associatedTokenAccountExists) {
  const createAssociatedTokenAccountInstruction = await Token.createAssociatedTokenAccountInstruction(
    ASSOCIATED_TOKEN_PROGRAM_ID,
    TOKEN_PROGRAM_ID,
    new PublicKey("<TOKEN_MINT_ADDRESS>"),
    associatedTokenAddress,
    payerAccount.publicKey,
    payerAccount.publicKey,
  );
  instructions.push(createAssociatedTokenAccountInstruction);
}

// Mint some tokens to the payer account
const mintAmount = 1000000; // 1 million tokens
const mintToInstruction = await token.createMintToInstruction(
  new PublicKey("<TOKEN_MINT_ADDRESS>"),
  tokenAccountAddress,
  payerAccount.publicKey,
  [],
  mintAmount,
);

// Add the mint to instruction to the transaction instructions
instructions.push(mintToInstruction);

// Create the transaction
const transaction = new Transaction({ feePayer: payerAccount.publicKey, instructions });

// Sign and send the transaction
const signedTransaction = await provider.wallet.signTransaction(transaction);
const transactionSignature = await provider.connection.sendRawTransaction(signedTransaction.serialize());
await provider.connection.confirmTransaction(transactionSignature);

File: Solana_Token.rs
use solana_client::{rpc_client::RpcClient, rpc_config::RpcSendTransactionConfig};
use solana_program::{
    instruction::{AccountMeta, Instruction},
    pubkey::Pubkey,
    system_instruction,
    system_program, 
    program_pack::Pack,
};
use spl_token::{
    self, 
    instruction::{initialize_account, initialize_mint, mint_to},
    state::{Account as TokenAccount, Mint},
};

fn main() {
    // Set the network and token account addresses
    let rpc_url = "https://api.mainnet-beta.solana.com".to_string();
    let token_account_address = Pubkey::from_str("<TOKEN_ACCOUNT_ADDRESS>").unwrap();

    // Set the payer account and its associated token account
    let payer_account = solana_sdk::signature::Keypair::from_bytes(&hex::decode("<PAYER_PRIVATE_KEY>").unwrap()).unwrap();
    let associated_token_address = spl_token::state::Account::get_associated_address(
        &payer_account.pubkey(),
        &spl_token::id(),
        &Pubkey::from_str("<TOKEN_MINT_ADDRESS>").unwrap(),
    );

    // Connect to the Solana RPC endpoint
    let rpc_client = RpcClient::new(rpc_url);

    // Create the token object
    let token_program_id = spl_token::id();
    let token = Token::new(&rpc_client, Pubkey::from_str("<TOKEN_MINT_ADDRESS>").unwrap());

    // Create the transaction instructions
    let mut instructions = vec![];

    // Create the token account if it doesn't exist
    let token_account_exists = match rpc_client.get_account(&token_account_address) {
        Ok(Some(_)) => true,
        _ => false,
    };
    if !token_account_exists {
        let create_account_instruction = system_instruction::create_account(
            &payer_account.pubkey(),
            &token_account_address,
            1.max(token_account_layout.span()),
            token_account_layout.span(),
            &token_program_id,
        );
        instructions.push(create_account_instruction);
    }

    // Create the associated token account if it doesn't exist
    let associated_token_account_exists = match rpc_client.get_account(&associated_token_address) {
        Ok(Some(_)) => true,
        _ => false,
    };
    if !associated_token_account_exists {
        let create_associated_token_account_instruction = spl_token::instruction::create_associated_token_account(
            &payer_account.pubkey(),
            &payer_account.pubkey(),
            &Pubkey::from_str("<TOKEN_MINT_ADDRESS>").unwrap(),
            &associated_token_address,
        );
        instructions.push(create_associated_token_account_instruction);
    }

    // Mint some tokens to the payer account
    let mint_amount = 1_000_000; // 1 million tokens
    let mint_to_instruction = mint_to(
        &token_program_id,
        &Pubkey::from_str("<TOKEN_MINT_ADDRESS>").unwrap(),
        &token_account_address,
        &payer_account.pubkey(),
        &[&payer_account.pubkey()],
        mint_amount,
    );
    instructions.push(mint_to_instruction);

    // Create the transaction
    let transaction = Instruction {
        program_id: system_program::id(),
        accounts: vec![],
        data: vec![],
    }
    .with_instructions(instructions);
    let recent_blockhash = rpc_client.get_recent_blockhash().unwrap().0;
    let mut transaction_signatures = rpc_client
        .send_transaction_with_config(
            &transaction,
            RpcSendTransactionConfig {
                preflight_commitment: None,
                skip_preflight: false,
                preflight_timeout: None,
